### PR TITLE
Feature: create search params hooks

### PR DESCRIPTION
**Pull request by alnavarrop99**
**Description**:
Hook for reading the search parameters of the route as an argument or of the current route.

**Uses Cases**:
- ```useSearchParams()```: current url search params.
- ```useSearchParams("https://example.com?foo=1&bar=2&foo=2")```: foo: 1, bar: 2.
- ```useSearchParams("https://example.com?foo=1&bar=2&foo=2", { unique: false })```: foo: [1,2], bar: 2.

_Examples_: 
```tsx
import { useSearchParams } from "./hooks/ts/useSearchParams";

const App = () => {
  const params = useSearchParams("https://example.com?foo=1&bar=2&foo=2", { unique: true })
  return <div>
    <p>Typescript test:</p>
    { Object.entries(params)?.map( ([key, value]) => (
      <p key={key}>
        <span>{key}</span>: <span>{value}</span>
      </p>
    ) ) }
  </div>
};

export default App;
```

### CODE REVIEW

**Comment by alnavarrop99** [comment]
Do you also think it's necessary to include search parameter modification methods like `has`, `set`, `append`, `delete` that interact directly with the route? I'm not sure if it's a good idea since in most cases the project's router handles this, and I'm not completely sure if there could be incompatibilities between the hook and the router. [Moxilla](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append)

**Review by dlcastillop — CHANGES_REQUESTED** [review]
Update your branch with the latest changes in the repo. Now, you only must provide the hook in TypeScript (like you sugested 🙌) and all hooks are in the `hooks` directory.

**Comment by dlcastillop** [comment]
> Do you also think it's necessary to include search parameter modification methods like `has`, `set`, `append`, `delete` that interact directly with the route? I'm not sure if it's a good idea since in most cases the project's router handles this, and I'm not completely sure if there could be incompatibilities between the hook and the router. [Moxilla](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append)

I do not think we must include it because like you said normally the project's router handles this.

File: src/hooks/ts/useSearchParams.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react"
+
+type TUseSearchParams = ( url?: string, opt?: { unique: boolean } ) => Record<string, any>
+export const useSearchParams: TUseSearchParams = ( url = location.search, opt = { unique: true } ) => {
+  const _urlSearch = new URL(url)
```
Comment (dlcastillop) [inline src/hooks/ts/useSearchParams.ts:5]: When I use the hook without specify an url this error ocurrs:

useSearchParams.ts:5 Uncaught TypeError: Failed to construct 'URL': Invalid URL at useSearchParams (useSearchParams.ts:5:22)

```ts
import { useSearchParams } from "./hooks/ts/useSearchParams";

const AppTs = () => {
  const params = useSearchParams();
  return (
    <div>
      <p>Typescript test:</p>
      {Object.entries(params)?.map(([key, value]) => (
        <p key={key}>
          <span>{key}</span>: <span>{value}</span>
        </p>
      ))}
    </div>
  );
};

export default AppTs;
```

**Review by dlcastillop — CHANGES_REQUESTED** [review]

**Comment by alnavarrop99** [comment]
Ready, I have resolved the conflicts and corrected the current path error.

File: src/hooks/useSearchParams.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react"
+
+type TUseSearchParams = ( url?: string, opt?: { unique: boolean } ) => Record<string, any>
```
Comment (dlcastillop) [inline src/hooks/useSearchParams.ts:3]: I am also getting a lot of problems with the types

image

Does not your linter show it to you?

**Review by dlcastillop — CHANGES_REQUESTED** [review]

**Comment by dlcastillop** [comment]
Great work @alnavarrop99! 🙌

**Pull request merged**